crossing-repair.lic

```diff
@@ -2,37 +2,184 @@
+    # repair type is specified for the target town.
+    leather_repairer = @hometown_data['leather_repair'] || @hometown_data['metal_repair']
+    metal_repairer = @hometown_data['metal_repair'] || @hometown_data['leather_repair']
+
+    # If we have no currency listed for our target town in base-town, we can't do anything, not even
+    # check if we have enough coin to exceed our yaml-specified repair_withdrawl_amount, since
+    # we don't know the requisite currency for the repair shop! Otherwise, if we know the currency,
+    # check if we have sell_loot_skip_bank set in our yaml AND we have enough coin on hand to exceed
+    # our yaml specifications. If so, don't care whether a bank is specified for this town.
+    # Otherwise we must do banking, and need the appropriate checks.
+    if !@hometown_data['currency']
+      DRC.message("No known currency for target town. Exiting.")
+      DRC.message("To add a currency for this town, please submit an issue on GitHub:")
+      DRC.message("https://github.com/rpherbig/dr-scripts/issues")
+      exit
+    else
+      if DRCM.wealth(settings['hometown']) >= @repair_withdrawal_amount && @skip_bank
+        DRC.message("We have enough money and our yaml specifies not to deposit coins after repair.") if $debug_mode_crossing_repair
+        DRC.message("Continuing...") if $debug_mode_crossing_repair
+      else
+        DRC.message("Either we don't have enough money on hand, or we don't specify sell_loot_skip_bank.") if $debug_mode_crossing_repair
+        DRC.message("Must bank...") if $debug_mode_crossing_repair
+        if @hometown_data['deposit']
+          DRCM.ensure_copper_on_hand(@repair_withdrawal_amount, settings)
+        else
+          DRC.message("No bank specified for this town, AND we don't have enough money on hand,")
+          DRC.message("and/or we have not specified to skip depositing excess coins after repair.")
+          DRC.message("To add a bank for this town, please submit an issue on GitHub:")
+          DRC.message("https://github.com/rpherbig/dr-scripts/issues")
+          exit
+        end
+      end
+    end
+
+    # If we have no repairers specified for this town, message user to specify a different town.
+    # If they believe a repairer should exist for this town, direct them to submit an issue. Exit.
+    unless leather_repairer || metal_repairer
+      DRC.message("No repairers found for #{town} in base-town.yaml. Please choose a different town.")
+      DRC.message("To add repairers for this town, please submit an issue on GitHub:")
+      DRC.message("https://github.com/rpherbig/dr-scripts/issues")
+      exit
+    end
+
+    # If we've not exited above, we're validated ready to repair.
+    # Populate a hash of items to repair at either leather or metal repairer, or both.
+    repair_info = { leather_repairer => leather_items_to_repair }
+                    .merge(metal_repairer => metal_items_to_repair) { |_key, old, new| old + new }
```

Comment:
🧹  There's just something about the way this merge statement is done with the hashes being defined in the same line that just doesn't read well for me. Stylistically, I vote for the following, noting that it's six of one, half-dozen of another.
```ruby
    leather_repair_info = { leather_repairer => leather_items_to_repair }
    metal_repair_info = { metal_repairer => metal_items_to_repair }
    repair_info = leather_repair_info.merge(metal_repair_info) { |key, old, new| old + new }
```